vm/systemSmartContracts/delegation.go

```diff
@@ -377,6 +393,40 @@ func (d *delegation) delegateUser(
 	return vmcommon.Ok
 }
 
+func (d *delegation) makeStakeArgsIfAutomaticActivation(
+	config *DelegationConfig,
+	status *DelegationContractStatus,
+	globalFund *GlobalFundData,
+) [][]byte {
+	lenStakableKeys := uint64(len(status.NotStakedKeys)) + uint64(len(status.UnStakedKeys))
+	if !config.AutomaticActivation || lenStakableKeys == 0 {
+		return nil
+	}
+
+	maxNodesToStake := big.NewInt(0).Div(globalFund.TotalActive, d.nodePrice).Uint64()
+	numStakedNodes := uint64(len(status.StakedKeys))
+	if maxNodesToStake <= numStakedNodes {
+		return nil
+	}
+
+	numNodesToStake := maxNodesToStake - numStakedNodes
+	gasLeftToStakeNumNodes := d.eei.GasLeft() / d.gasCost.MetaChainSystemSCsCost.Stake
+
+	numNodesToStake = core.MinUint64(core.MinUint64(lenStakableKeys, numNodesToStake), gasLeftToStakeNumNodes)
+	if numNodesToStake == 0 {
+		return nil
+	}
+
+	stakeArgs := [][]byte{big.NewInt(0).SetUint64(numNodesToStake).Bytes()}
+	listOfStakableNodes := append(status.NotStakedKeys, status.UnStakedKeys...)
```

Comment:
it is not someBody - the stakingProvider unStaked the nodes - if he wants to keep an order will set automatic activation to false. thus this will not enter. The other way is for him to add more keys thus noone will take from UnStakedKeys.

Reply:
ok